README.md

```diff
@@ -5,757 +5,875 @@
+
+## Explicit Transitions vs Transition Matching
+
+Most state machine libraries focus on finding the next state given configuration. For example, [xstate](https://github.com/davidkpiano/xstate#finite-state-machines) declaration describes what state id to match when in a specific state.
 
 ```js
-today.todos[0].title.set('Buy 2% milk').state;
-// => { todos: [ Todo { title: 'Buy 2% milk', isCompleted: false } ] }
+import { Machine } from 'xstate';
 
-today.todos.push({ title: 'Review PRs' }).state
-// => { todos: [ Todo { title: 'Buy 2% milk', isCompleted: false }, Todo { title: 'Review PRs', isCompleted: false } ] }
+const lightMachine = Machine({
+  key: 'light',
+  initial: 'green',
+  states: {
+    green: {
+      on: {
+        TIMER: 'yellow',
+      }
+    },
+    yellow: {
+      on: {
+        TIMER: 'red',
+      }
+    },
+    red: {
+      on: {
+        TIMER: 'green',
+      }
+    }
+  }
+});
 ```
 
-You can do the same with Objects using `{Type}` syntax. 
+Microstates does not do any special state resolution. You explicitly declare what happens on a state transition. Here is what a similar state machine looks like in Microstates.
 
 ```js
-let today = create({Todo}, { t1: { title: 'Buy milk' }, t2: { title: 'Review PRs' } }).state
-// => { t1: Todo { title: 'Buy milk', isComplete: false }, t2: Todo { title: 'Review PRs', isComplete: false } }
+class LightMachine {
+  color = String;
+
+  initialize({ color: 'green' } = {}) {
+    return this.color.set(color);
+  }
 
-today.t2.isComplete.toggle().state
-// => { t1: Todo { title: 'Buy milk', isComplete: false }, t2: Todo { title: 'Review PRs', isComplete: true } }
+  timer() {
+    switch (this.color.state) {
+      case 'green': return this.color.set('yellow');
+      case 'yellow': return this.color.set('red');
+      case 'red':
+      default:
+        return this.color.set('green'); 
+    }
+  }
+}
 ```
 
-## Observable interoperability
+With Microstates, you explicitely describe what happens on transition and define the matching mechanism. 
 
-You can create an observable from any microstate using the `Observable.from` method.
-The resulting Observable will stream the next microstate for every transition.
-When you subscribe to the observable microstate, you'll immediately receive a microstate through the stream.
-This microstate will have transitions that you can call to cause the next microstate to come through the stream.
+### No transitionTo function
 
-```js
-import { create } from 'microstates';
-import { Observable } from 'rxjs';
+`transitionTo` is often used by state machine libraries to trigger state transition. Here is an example with xstate library,
 
-let ms = create(Number, 42);
-let observable = Observable.from(ms);
-
-let last;
-
-observable.subscribe(microstate => {
-  last = microstate;
-  console.log(microstate.state);
-});
-// => 42
+```js
+const nextState = lightMachine.transition('green', 'TIMER').value;
```

Comment:
In a way, microstates does have a transitionTo method. It's called "vanilla JavaScript property lookup"

```js
let next = lightMachine['timer']();
```

Reply:
Yeah, how about this?

> Microstates does not have such a method. Instead, it relies on vanilla JavaScript property lookup. The method invocation is equivalent to calling `transitionTo` with name of the transition.

Reply:
I love it.